auth/utils/restconfig.go

```diff
@@ -84,11 +101,20 @@ func GetRESTConfig(ctx context.Context,
 	}
 
 	// Get provider by name.
-	provider, err := ProviderByName[auth.RESTConfigProvider](cm.Data[meta.KubeConfigKeyProvider])
+	providerName := cm.Data[meta.KubeConfigKeyProvider]
+	provider, err := ProviderByName[auth.RESTConfigProvider](providerName)
```

Comment:
`ProviderByName` should handle creating the provider

```suggestion
	provider, err := ProviderByName[auth.RESTConfigProvider](providerName, kubeClient)
```